src/services/aws_kms/mod.rs

```diff
@@ -0,0 +1,385 @@
+            ))?
+            .into_inner();
+
+        Ok(der_pk_blob)
+    }
+
+    async fn sign_digest<'a, 'b>(
+        &'a self,
+        key_id: &'b str,
+        digest: [u8; 32],
+    ) -> AwsKmsResult<Vec<u8>> {
+        // Sign the digest with the AWS KMS
+        let sign_result = self
+            .inner
+            .sign()
+            .key_id(key_id)
+            .signing_algorithm(SigningAlgorithmSpec::EcdsaSha256)
+            .message_type(MessageType::Digest)
+            .message(Blob::new(digest))
+            .send()
+            .await;
+
+        // Process the result, extract DER signature
+        let der_signature = sign_result
+            .map_err(|e| AwsKmsError::PermissionError(e.to_string()))?
+            .signature
+            .ok_or(AwsKmsError::SignError(
+                "Signature not found in response".to_string(),
+            ))?
+            .into_inner();
+
+        Ok(der_signature)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AwsKmsService<T: AwsKmsClientTrait + Clone = AwsKmsClient> {
+    pub kms_key_id: String,
+    client: T,
+}
+
+impl AwsKmsService<AwsKmsClient> {
+    pub async fn new(config: AwsKmsSignerConfig) -> AwsKmsResult<Self> {
+        let region_provider =
+            RegionProviderChain::first_try(config.region.map(Region::new)).or_default_provider();
+
+        let auth_config = aws_config::defaults(BehaviorVersion::latest())
+            .region(region_provider)
+            .load()
+            .await;
+        let client = AwsKmsClient {
+            inner: Client::new(&auth_config),
+        };
+
+        Ok(Self {
+            kms_key_id: config.key_id,
+            client,
+        })
+    }
+}
+
+#[cfg(test)]
+impl<T: AwsKmsClientTrait + Clone> AwsKmsService<T> {
+    pub fn new_for_testing(client: T, config: AwsKmsSignerConfig) -> Self {
+        Self {
+            client,
+            kms_key_id: config.key_id,
+        }
+    }
+}
+
+impl<T: AwsKmsClientTrait + Clone> AwsKmsService<T> {
+    /// Signs a bytes with the private key stored in AWS KMS.
+    ///
+    /// Pre-hashes the message with keccak256.
+    pub async fn sign_bytes_evm(&self, bytes: &[u8]) -> AwsKmsResult<Vec<u8>> {
+        // Create a digest of a message payload
+        let digest = keccak256(bytes).0;
+
+        // Sign the digest with the AWS KMS
+        // Process the result, extract DER signature
+        let der_signature = self.client.sign_digest(&self.kms_key_id, digest).await?;
+
+        // Parse DER into Secp256k1 format
+        let rs = k256::ecdsa::Signature::from_der(&der_signature)
+            .map_err(|e| AwsKmsError::ParseError(e.to_string()))?;
+
+        let der_pk = self.client.get_der_public_key(&self.kms_key_id).await?;
+
+        // Extract public key from AWS KMS and convert it to an uncompressed 64 pk
+        let pk = extract_public_key_from_der(&der_pk)
+            .map_err(|e| AwsKmsError::ParseError(e.to_string()))?;
```

Comment:
This is a matter of taste, but I prefer not to have the same error to appear in two different calls.
The reason for this is when you make a release build, if some occasional unwrap on error happens you will not receive the line where it has happened. The error that was unwrapped and logs that were collected will be the only two things you will have to debug the issue.